src/audio/tdfb/tdfb.c

```diff
@@ -31,6 +31,12 @@
 #include <stddef.h>
 #include <stdint.h>
 
+/* The driver assigns running numbers for control index. The enum controls
+ * get indices 0 and 1. Make sure the on/off control is first in topology.
+ */
+#define ENUM_CTRL_ONOFF		0
```

Comment:
Can we not assign an ID to each control as using index is easily broken ? @juimonen ?

Reply:
I will need to remove enum controls add from topology to make this work with current kernel. I'd like to leave this to FW though.  It's best that @juimonen proposes how to do, e.g. is this approach seen in this PR OK.

Reply:
OK to resolve?

Reply:
What is the kernel limitation?